adapter/pkg/subscription/subscription.go

```diff
@@ -79,33 +84,54 @@ var (
 			responseType: SubPolicyList,
 		},
 	}
-	accessToken string
-	conf        *config.Config
+	// APIListChannel is used to add apis
+	APIListChannel chan response
+	accessToken    string
+	conf           *config.Config
 )
 
 type response struct {
-	Error    error
-	Payload  []byte
-	Endpoint string
-	Type     interface{}
+	Error        error
+	Payload      []byte
+	Endpoint     string
+	GatewayLabel string
+	Type         interface{}
 }
 
 type resource struct {
 	endpoint     string
 	responseType interface{}
 }
 
+func init() {
+	APIListChannel = make(chan response)
+	APIListMap = make(map[string]*resourceTypes.APIList)
+}
+
 // LoadSubscriptionData loads subscription data from control-plane
 func LoadSubscriptionData(configFile *config.Config) {
 	conf = configFile
 	accessToken = auth.GetBasicAuth(configFile.ControlPlane.EventHub.Username, configFile.ControlPlane.EventHub.Password)
 
 	var responseChannel = make(chan response)
 	for _, url := range resources {
-		go invokeService(url.endpoint, url.responseType, responseChannel, 0)
+		go InvokeService(url.endpoint, url.responseType, nil, responseChannel, 0)
+	}
+
+	// Take the configured labels from the adapter
+	configuredEnvs := conf.ControlPlane.EventHub.EnvironmentLabels
+
+	if len(configuredEnvs) > 0 {
```

Comment:
If condition can be removed

Reply:
fixed via https://github.com/wso2/product-microgateway/pull/1651/commits/34135e565e0feb892396850af240c35fdc8c4499